### PR TITLE
2949 ontogeny factor table and ontogeny factor gi table molecule or expression property

### LINKED ISSUE #2949: `Ontogeny factor table` and `Ontogeny factor GI table` - molecule or expression property?
The `Ontogeny factor table` and the `Ontogeny factor GI table` are now present in different places.

- Directly in the molecule, where they are not tables but constant values. When exporting from PK-Sim, I assume that the first value is the table is taken, so the value in the molecule-BB for CYP3A4 is 1e-3:

image

- In the expression profile:

image

When creating a simulation WITH the expression profile, the correct values are applied because the parameter is overwritten from the expression BB:

image

But it is also possible to create a simulation without the Expression, e.g., when the spatial structure has been exported and the expression profile has been selected to create the Parameter Values BB. In this case, the ontogeny table is not updated but the value from the molecules BB is used:

image

And the value is obviously wrong.

To fix this, I propose the following:

- [ ] `Ontogeny factor table` and the `Ontogeny factor GI table` in the Molecules BB are ALWAYS 1
- [ ] Exporting from a spatial structure container and selecting an expression profiles also creates the entries for the ontogeny tables in the PV BBs of the export (disregarding of what organ is exported)
- [ ] PV BB: "Add Protein Expression": also create the entries for the ontogeny tables from the selected expression.

### PR DESCRIPTION
**Pull request by msevestre**
Fixes #2949 

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [ ] Unit tests
- [ ] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: PKSim.sln [hunk 1]
```diff
@@ -36,8 +35,6 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "PKSim.Assets", "src\PKSim.A
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "PKSim.Infrastructure", "src\PKSim.Infrastructure\PKSim.Infrastructure.csproj", "{4A55E410-29A7-4BD7-A731-09A9F04BFEFC}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "PKSim.Matlab", "src\PKSim.Matlab\PKSim.Matlab.csproj", "{2F19A0C6-4C93-4A78-8DB6-E4C3EDE8E6F6}"
```
Comment (msevestre) [inline PKSim.sln:39]: unrelated but removed the matlab stuff that is not used anymore

File: src/Db/Dump/PKSimDB_dump.txt [hunk 2]
```diff
@@ -1207,6 +1207,7 @@ INSERT INTO tab_categories VALUES('VascularEndotheliumVolume','Volume of vascula
 INSERT INTO tab_categories VALUES('ExpressionParameters','Expression parameters','Expression parameters','Individual',NULL);
 INSERT INTO tab_categories VALUES('DiseaseStates','Disease States','Disease state parameters','Individual',NULL);
 INSERT INTO tab_categories VALUES('Individual_AgeDependent','Individual age dependent','Individual age dependent','Individual','Individual');
+INSERT INTO tab_categories VALUES('OntogenyFactors','Ontogeny factors','Ontogeny factors','Individual','Individual');
```
Comment (msevestre) [inline src/Db/Dump/PKSimDB_dump.txt:1210]: @Yuri05 New category and calculation method
Idea: OntogenyFactors CM is only used in the Expression Profile. Individual etc do not use it. That way, I am only generating the formula when required

File: src/Db/Dump/PKSimDB_dump.txt [hunk 3]
```diff
@@ -3748,8 +3750,8 @@ INSERT INTO tab_calculation_method_rates VALUES('DiseaseStates','One_Rate','1','
 INSERT INTO tab_calculation_method_rates VALUES('Events','Urinary_Bladder_Emptying_StartEvent_Condition','UrinaryBladderEmptyingEnabled AND (Time=StartTime)','Dimensionless');
 INSERT INTO tab_calculation_method_rates VALUES('DynamicSumFormulas','PARAM_VolumePlasma','V_#i','Volume');
 INSERT INTO tab_calculation_method_rates VALUES('DiseaseStates','PARAM_ChildPughScore','0','Dimensionless');
-INSERT INTO tab_calculation_method_rates VALUES('ExpressionParameters','TableFormulaWithXArgument_OntogenyFactor',NULL,'Dimensionless');
-INSERT INTO tab_calculation_method_rates VALUES('ExpressionParameters','TableFormulaWithXArgument_OntogenyFactorGI',NULL,'Dimensionless');
+INSERT INTO tab_calculation_method_rates VALUES('OntogenyFactors','TableFormulaWithXArgument_OntogenyFactor',NULL,'Dimensionless');
```
Comment (msevestre) [inline src/Db/Dump/PKSimDB_dump.txt:3753]: ExpressionParameters are used all the time for expression so the table was also created all the time. Now I am only creating it for the right species

File: tests/PKSim.UI.Tests/PKSim.UI.Tests.csproj [hunk 4]
```diff
@@ -48,7 +48,7 @@
 		<ItemGroup>
 			<WindowsFiles Include="$(TargetDir)x64/SQLite.Interop.dll" />
 		</ItemGroup>
-		<Copy SourceFiles="@(WindowsFiles);" DestinationFolder="$(TargetFolder)" DestinationFiles="@(WindowsFiles->Replace('x64/SQLite.Interop.dll', 'SQLite.Interop.dll'))" />
+		<Copy SourceFiles="@(WindowsFiles);" DestinationFolder="$(TargetFolder)" DestinationFiles="@(WindowsFiles-&gt;Replace('x64/SQLite.Interop.dll', 'SQLite.Interop.dll'))" />
```
Comment (msevestre) [inline tests/PKSim.UI.Tests/PKSim.UI.Tests.csproj:51]: @rwmcintosh Any idea if these changes are legit? This was done automatically when I updated nuget packages

**Review by msevestre — COMMENTED** [review]
@Yuri05 It took a right to get all the tests passing again. I am assuming that we are covering ALL possible scenarios with our tests for different species and enzyme etc so I am not going to write singular tests unless the behavior is new.

File: src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
+using DevExpress.Data.Controls.ExpressionEditor;
```
[the remark this answers is not in the document]
Comment (msevestre) [reply src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs:3]: we don't. Good catch

[the remark this answers is not in the document]
Comment (msevestre) [reply src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs:3]: added automaticaly

**Review by Yuri05 — APPROVED** [review]

**Comment by Yuri05** [comment]
Looks good for me. However, some tests fail now (population creation from R) ...

**Comment by msevestre** [comment]
Interesting.  Why would R fail now lol

**Comment by msevestre** [comment]
ok just updated the code. Should be good now

**Pull request merged**